src/meshio/abaqus/_abaqus.py

```diff
@@ -249,24 +250,28 @@ def _read_cells(f, params_map, point_ids):
         raise ReadError(f"Element type not available: {etype}")
 
     cell_type = abaqus_to_meshio_type[etype]
+    # ElementID + NodesIDs
+    num_data = num_nodes_per_cell[cell_type] + 1
 
     cells, idx = [], []
     cell_ids = {}
-    counter = 0
     while True:
         line = f.readline()
         if not line or line.startswith("*"):
             break
-        if line.strip() == "":
-            continue
-
         line = line.strip()
+        if line == "":
+            continue
         idx += [int(k) for k in filter(None, line.split(","))]
-        if not line.endswith(","):
-            cell_ids[idx[0]] = counter
-            cells.append([point_ids[k] for k in idx[1:]])
-            idx = []
-            counter += 1
+
+    # Check for expected number of data
+    if len(idx) % num_data != 0:
+        raise ReadError("Expected number of data items does not match element type")
+
```

Comment:
Can't we just `idx = numpy.array(idx)` here and then pick the first column as id and the therest as point_ids?

Reply:
I changed that now, but I'm not sure if that is better.
The issue is that we have to translate the node ids and I do not know if there is a better way than using the list comprehension

Reply:
What about
```python
cells = point_ids[idx[:, 1:]]
```
?

Reply:
The `cells_ids` comprehensions looks weird, too. They could be handled with `dict(zip())`.

Reply:
```
cells = point_ids[idx[:, 1:]]
```
does not work because point_ids is a `dict` and `idx` is `numpy.ndarray`.